### PR TITLE
Tracing

### PR DESCRIPTION
**Pull request by elijahbenizzy**
| <a href="https://ellipsis.dev" target="_blank">Ellipsis</a> | :rocket: This PR description was created by [Ellipsis](https://www.ellipsis.dev) for commit 8de0304513b4411d2706003fa6627f0e32ef2a59.  | 
|--------|--------|

### Summary:
This PR introduces a new tracing feature, updates classes for asynchronous actions, introduces new models, modifies the UI layout, updates UI components, adds new chip colors, introduces a new `get_uri` function, simplifies the code, fixes a key issue, and corrects text.

**Key points**:
- Introduced new tracing feature with `TracerFactory` class.
- Updated `Application` class for asynchronous actions.
- Introduced new models: `BeginEntryModel`, `BeginSpanModel`, `EndEntryModel`, `Span`, and `Step`.
- Modified UI layout to include a 'minimize' feature.
- Added new chip colors for 'demo' and 'test'.
- Updated `StepList` and `GraphView` UI components to display new tracing feature.
- Added `get_uri` function to map project IDs to URIs.
- Simplified code by removing use of `TracerFactory`.
- Fixed key issue in `TableRow` component in `AppList.tsx`.
- Corrected text in `/burr/cli/demo_data.py`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Comment by elijahbenizzy** [comment]
See https://github.com/DAGWorks-Inc/burr/issues/51

**Comment by elijahbenizzy** [comment]
TODO for implementation:
- [ ] Implement basic context manager
- [ ] Implement injection of parameters (don't expose yet)
- [ ] Implement hooks
- [ ] Test with hooks
- [ ] Add more unit tests
- [ ] Integrate with Tracker
- [ ] Integerate with UI

File: telemetry/ui/src/components/routes/app/StepList.tsx [hunk 1]
```diff
@@ -41,30 +53,216 @@ const AutoRefreshSwitch = (props: {
   autoRefresh: boolean;
   setAutoRefresh: (b: boolean) => void;
 }) => {
+  const AutoRefreshIcon = props.autoRefresh ? PauseIcon : ArrowPathIcon;
```
[the remark this answers is not in the document]
Comment (elijahbenizzy) [reply telemetry/ui/src/components/routes/app/StepList.tsx:56]: I really think ternary expressions are much cleaner than a map of `true`/`false` keys, no?

**Comment by elijahbenizzy** [comment]
Note this will make current logs break, and I'm ok with that for now.

File: docs/concepts/additional-visibility.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+You can make observations on the state by calling out to the `log_artifact` method on the `__tracer` context manager.
+For instance:
+
+.. code-block:: python
+
+    from burr.visibility import TracingFactory
+    from burr.core import action
+
+    @action(reads=['input_var'], writes=['output_var'], __tracer: TracingFactory)
```
Comment (elijahbenizzy) [inline docs/concepts/additional-visibility.rst:74]: Wrong -- not valid python

File: docs/concepts/additional-visibility.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    def my_action(state: State) -> Tuple[dict, State]:
```
Comment (elijahbenizzy) [inline docs/concepts/additional-visibility.rst:75]: Add `__logger` to demonstrate

File: telemetry/ui/src/components/routes/ProjectList.tsx [hunk 4]
```diff
@@ -21,31 +23,56 @@ export const ProjectListTable = (props: { projects: Project[] }) => {
...
-            </TableCell> */}
-            <TableCell>
-              <Button color="white" href={`/project/${project.id}`}>
-                App Runs
-              </Button>
-            </TableCell>
-          </TableRow>
-        ))}
+        {projectsSorted.map((project) => {
+          let projectName = project.name;
```
Comment (elijahbenizzy) [inline telemetry/ui/src/components/routes/ProjectList.tsx:33]: Link to project appears to be busted

**Review by elijahbenizzy — COMMENTED** [review]
Needs a few things

**Review by skrawcz — APPROVED** [review]

**Pull request merged**